src/Entity/SmsDeliveryReport.php

```diff
@@ -89,30 +96,64 @@ public function setStatusMessage($message) {
   /**
    * {@inheritdoc}
    */
+  public function getStatusTime() {
+    return $this->get('status_time')->value;
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function setStatusTime($timestamp) {
+    $this->set('status_time', $timestamp);
+    return $this;
+  }
+
+  /**
+   * {@inheritdoc}
+   */
   public function getTimeQueued() {
-    return $this->get('time_queued')->value;
+    $queued = $this->getRevisionAtStatus(SmsMessageReportStatus::QUEUED);
+    return $queued ? $queued->getStatusTime() : NULL;
   }
 
   /**
    * {@inheritdoc}
    */
   public function setTimeQueued($time) {
```

Comment:
This makes sense if there was no previous revision at that status. But what if there is already a revision at that status? We end up with multiple revisions that have the same status, which will also be tricky when retrieving the revision that has a particular status.

Or maybe throw an exception??